build_scripts/api/helpers/image_builder.py

```
@@ -0,0 +1,271 @@
+
+            self.image_pipe.to(accelerator.device)
+            new_hotness = os.path.join(config.model_dir, "checkpoints", f"checkpoint-{config.revision}")
+            if os.path.exists(new_hotness):
+                accelerator.print(f"Resuming from checkpoint {new_hotness}")
+                disable_safe_unpickle()
+                accelerator.load_state(new_hotness)
+                enable_safe_unpickle()
+
+            if config.use_lora and lora_model:
+                lora_model_path = shared.ui_lora_models_path
+                if os.path.exists(lora_model_path):
+                    patch_pipe(
+                        pipe=self.image_pipe,
+                        maybe_unet_path=lora_model_path,
+                        unet_target_replace_module=get_target_module("module", config.use_lora_extended),
+                        token=None,
+                        r=lora_unet_rank,
+                        r_txt=lora_txt_rank
+                    )
+                    tune_lora_scale(self.image_pipe.unet, config.lora_weight)
+
+                    lora_txt_path = _text_lora_path_ui(lora_model_path)
+                    if os.path.exists(lora_txt_path):
+                        tune_lora_scale(self.image_pipe.text_encoder, config.lora_txt_weight)
+
+        else:
+            try:
+                from modules import sd_models
+                current_model = sd_models.select_checkpoint()
+                print(f"Source checkpoint: {source_checkpoint}")
+                new_model_info = get_checkpoint_match(source_checkpoint)
+                print(f"Model info: {new_model_info.filename}")
+                self.last_model = current_model
+                if new_model_info is not None:
+                    print(f"Loading model: {new_model_info.model_name}")
+                    shared.sd_model = sd_models.load_model(new_model_info)
+                    reload_system_models()
+            except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```